vulnerabilities/importers/debian.py

```diff
@@ -21,115 +21,194 @@
 
-import dataclasses
+import logging
 from typing import Any
+from typing import Iterable
 from typing import List
 from typing import Mapping
-from typing import Set
 
 import requests
-from dateutil import parser as dateparser
+from django.db.models.query import QuerySet
 from packageurl import PackageURL
+from univers.version_range import DebianVersionRange
+from univers.versions import DebianVersion
 
+from vulnerabilities.helpers import AffectedPackage as LegacyAffectedPackage
+from vulnerabilities.helpers import get_affected_packages_by_patched_package
+from vulnerabilities.helpers import get_item
 from vulnerabilities.helpers import nearest_patched_package
 from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
 from vulnerabilities.importer import Importer
 from vulnerabilities.importer import Reference
+from vulnerabilities.importer import UnMergeablePackageError
+from vulnerabilities.improver import MAX_CONFIDENCE
+from vulnerabilities.improver import Improver
+from vulnerabilities.improver import Inference
+from vulnerabilities.models import Advisory
 
+logger = logging.getLogger(__name__)
 
-class DebianImporter(Importer):
-    def __enter__(self):
-        if self.response_is_new():
-            self._api_response = self._fetch()
+IGNORABLE_VERSIONS = {"3.8.20-4."}
 
-        else:
-            self._api_response = {}
 
-    def updated_advisories(self) -> Set[AdvisoryData]:
-        advisories = []
+class DebianImporter(Importer):
 
-        for pkg_name, records in self._api_response.items():
-            advisories.extend(self._parse(pkg_name, records))
+    spdx_license_expression = "MIT"
+    license_url = "https://www.debian.org/license"
 
-        return self.batch_advisories(advisories)
+    api_url = "https://security-tracker.debian.org/tracker/data/json"
 
-    def _fetch(self) -> Mapping[str, Any]:
-        return requests.get(self.config.debian_tracker_url).json()
+    def get_response(self):
+        response = requests.get(self.api_url)
+        if response.status_code == 200:
+            return response.json()
+        raise Exception(
+            f"Failed to fetch data from {self.api_url!r} with status code: {response.status_code!r}"
+        )
 
-    def _parse(self, pkg_name: str, records: Mapping[str, Any]) -> List[AdvisoryData]:
-        advisories = []
-        ignored_versions = {"3.8.20-4."}
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        response = self.get_response()
+        for pkg_name, records in response.items():
+            yield from self.parse(pkg_name, records)
 
+    def parse(
+        self, pkg_name: str, records: Mapping[str, Any], ignored_versions=IGNORABLE_VERSIONS
+    ) -> Iterable[AdvisoryData]:
         for cve_id, record in records.items():
-            impacted_purls, resolved_purls = [], []
+            affected_versions, fixed_versions = [], []
```

Comment:
Multiple initialization on one line is never super readable IMHO
```suggestion
            affected_versions = []
            fixed_versions = [] 
```